src/AppBundle/Installer/EzCommerceCleanInstaller.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace AppBundle\Installer;
+
+use EzSystems\PlatformInstallerBundle\Installer\CoreInstaller;
+use Symfony\Component\Filesystem\Filesystem;
+
+class EzCommerceCleanInstaller extends CoreInstaller
+{
+    use InstallerCommandExecuteTrait;
+
+    public function importData()
+    {
+        $liveSchema = $this->db->getSchemaManager()->createSchema();
+        $dbPlatform = $this->db->getDatabasePlatform();
+
+        foreach (['kaliop_migrations', 'kaliop_migrations_contexts'] as $tableName) {
+            if (!$liveSchema->hasTable($tableName)) {
+                continue;
+            }
+            $this->db->exec($dbPlatform->getTruncateTableSQL($tableName));
+        }
+
+        $migrationCommands = [
+            'ezplatform:install ezplatform-ee-clean',
+            'doctrine:schema:update --dump-sql --force',
+            'kaliop:migration:migrate --path migrations/kaliop_migrations/users/201804161300_users.yml -n',
+            'kaliop:migration:migrate --path migrations/kaliop_migrations/content_types/201804161300_content_types.yml -n',
+            'kaliop:migration:migrate --path migrations/kaliop_migrations/sections/201804161300_sections.yml -n',
+            'kaliop:migration:migrate --path migrations/kaliop_migrations/content/Components/201804161300_components.yml -n',
+            'kaliop:migration:migrate --path migrations/kaliop_migrations/content/Catalog/201804161300_product_catalog.yml -n',
+            'kaliop:migration:migrate --path migrations/kaliop_migrations/roles/201804161300_roles.yml -n',
+            'silversolutions:switchdataprovider ez',
```

Comment:
Is this writing to config? If so it will break on p.sh right?

Reply:
it should not - the config is already modified, the only changes are db updates (product catalog fields) - but I will test it

Reply:
Hi Damian,
the last command "silversolutions:switchdataprovider ez" will modify app/config/ecommerce.yml to activate ez as dataprovider.

Reply:
yes, but result of this command is already in this PR ecommerce.yml file, which means the command will only update the db
```
bin/console silversolutions:switchdataprovider ez
Applying the change to all available product catalogs.
Old configuration reference not found in file ./app/config/ecommerce.yml: config_data_provider_econtent
Configuration file was not modified.
Accessing Ez API to modify Product Catalogs (Location ID: 119)
Root node to be set: 119
Root location of location ID: 119 was set to 119
------------------------------
Script finished.
```
I may be wrong, but looking at `\Silversolutions\Bundle\EshopBundle\Command\SwitchDataProviderCommand::changeDataProviderResourceConfig` shows that file_put_contents will be executed only if the file need to be changed (which is not in our case)

Reply:
> the last command "silversolutions:switchdataprovider ez" will modify app/config/ecommerce.yml to activate ez as dataprovider.
@damianz5 I think this will be an issue for eZ Platform Cloud. config is not part of the writable folders. Am I right?

Reply:
Ok is see.

We have added some logic to the ezcommerce-clean installer is this might/will break p.sh:
- after the migrations have been processed we need to collect some location ids 
- the location ids will be added to a yml file in app/config

How to deal with those kind of requirements in p.sh?

Reply:
Even if not exactly clean, we'll need to change it to be a mount*, and add inline doc on why so people can remove it once they have done install stage before pushing code. Which is essentially what @damianz5's is doing here trying to prepare repo to be ready for prod already.

\* https://github.com/ezsystems/ezplatform-ee-demo/blob/master/.platform.app.yaml#L81-L83